proxy/src/main/java/com/wavefront/agent/listeners/tracing/TracePortUnificationHandler.java

```diff
@@ -174,13 +174,17 @@ protected void processLine(final ChannelHandlerContext ctx, @Nonnull String mess
       // check whether error span tag exists.
       boolean sampleError = alwaysSampleErrors && object.getAnnotations().stream().anyMatch(
           t -> t.getKey().equals(ERROR_SPAN_TAG_KEY) && t.getValue().equals(ERROR_SPAN_TAG_VAL));
-      if (sampleError || sample(object)) {
-        handler.report(object);
-      }
+      report(object, sampleError);
+    }
+  }
+
+  protected void report(Span object, boolean sampleError) {
+    if (sampleError || sample(object)) {
```

Comment:
you can have this outside of the report method, report method do the work of only reporting. Also add a java doc for this method since its exposed to extending members.